### PR TITLE
feat(ingestion): add ingestion warning if event more than 23 hours in the future

### PR DESCRIPTION
**Pull request by xvello**
## Problem

We accept events in the future. Although they don't show up in the live events table, we had users confused by persons created in the future by these events.

We decided to not drop these events, but we're adding an ingestion warning if timestamps are more than 23h in the future, to help diagnose instrumentation issues.

image

## Changes

- Update `parseEventTimestamp` and its ingestion warning callback
- Create new `event_timestamp_in_future` ingestion warning and add it to the UI

## How did you test this code?

Unit testing + local run

### CODE REVIEW

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx [hunk 1]
```diff
@@ -73,6 +74,26 @@ const WARNING_TYPE_RENDERER = {
...
+        const details = warning.details as {
+            timestamp: string
+            sentAt: string
+            offset: string
+            now: string
+            result: string
+        }
+        return (
+            <>
+                Timestamp computed to <code>{details.result}</code> from the following input:
```
Comment (hazzadous) [inline frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:87]: Is there anywhere we can highlight how and why result is calculated the way that it is? Just a comment around time skew etc. I found it very confusing when I was looking at this first.

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx [hunk 2]
```diff
@@ -73,6 +74,26 @@ const WARNING_TYPE_RENDERER = {
...
+                <ul>
+                    {details.timestamp ? <li>timestamp: {details.timestamp}</li> : ''}
```
Comment (hazzadous) [inline frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:89]:
```suggestion
                    {details.timestamp ? <li>Client provided timestamp: {details.timestamp}</li> : ''}
```

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx [hunk 3]
```diff
@@ -73,6 +74,26 @@ const WARNING_TYPE_RENDERER = {
...
+                    {details.sentAt ? <li> sent_at: {details.sentAt}</li> : ''}
```
Comment (hazzadous) [inline frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:90]:
```suggestion
                    {details.sentAt ? <li>Client provided sent_at: {details.sentAt}</li> : ''}
```

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx [hunk 4]
```diff
@@ -73,6 +74,26 @@ const WARNING_TYPE_RENDERER = {
...
+                    {details.offset ? <li>offset: {details.offset}</li> : ''}
+                    <li>capture time: {details.now}</li>
```
Comment (hazzadous) [inline frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:92]:
```suggestion
                    <li>PostHog server capture time: {details.now}</li>
```

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx (hunk 1 above)
Comment (xvello) [reply frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:87]: Definitely. We have an autogenerated `docs` links to `https://posthog.com/manual/data-management#an-event-was-sent-more-than-23-hours-in-the-future`, we need to contribute that documentation change. That's where we'll explain this.

Comment (hazzadous) [reply frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:87]: 👍

File: frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx [hunk 5]
```diff
@@ -73,6 +74,26 @@ const WARNING_TYPE_RENDERER = {
...
+            now: string
+            result: string
+        }
+        return (
+            <>
+                Timestamp computed to <code>{details.result}</code> from the following input:
+                <ul>
+                    {details.timestamp ? <li>Client provided timestamp: {details.timestamp}</li> : ''}
+                    {details.sentAt ? <li>Client provided sent_at: {details.sentAt}</li> : ''}
+                    {details.offset ? <li>offset: {details.offset}</li> : ''}
```
Comment (xvello) [inline frontend/src/scenes/data-management/ingestion-warnings/IngestionWarningsView.tsx:91]:
```suggestion
                    {details.offset ? <li>Client provided time offset: {details.offset}</li> : ''}
```

**Review by hazzadous — APPROVED** [review]

**Pull request merged**